### PR TITLE
Option to run 'dotnet test' with Azure CLI

### PR DESCRIPTION
**Pull request by chanw**

### CODE REVIEW

File: azure/pipelines/dotnet/ci.yml [hunk 1]
```diff
@@ -15,8 +15,12 @@ parameters:
 - name: DotNetSdkVersion
   type: string
   default: ''
-  displayName: 'Specifc .NET SDK version to use.'
+  displayName: 'Specific .NET SDK version to use.'
 
+- name: azureSubscription
+  type: string
+  default: ''
+  displayName: 'Optional Azure Subscription to run ''dotnet test''.'
```
Comment (WimVergouwe) [inline azure/pipelines/dotnet/ci.yml:23]: Explain you might want to use this.

File: azure/pipelines/dotnet/ci.yml [hunk 2]
```diff
@@ -168,13 +172,31 @@ steps:
...
           exit 0;
-  - task: DotNetCoreCLI@2
-    displayName: 'DotNet Test'
-    inputs:
-      command: 'test'
-      projects: '**/*[Tt]ests/*.csproj'
-      arguments: '--configuration ${{ parameters.BuildConfiguration }} --no-restore --no-build'
-    condition: succeededOrFailed()
+
+  - ${{ if ne(parameters.azureSubscription, '') }}:
```
Comment (WimVergouwe) [inline azure/pipelines/dotnet/ci.yml:176]: I suggest you put the 'regular' testing method first: condition: ${{ if eq(parameters.azureSubscription, '') }}

File: azure/pipelines/dotnet/ci.yml [hunk 3]
```diff
@@ -168,13 +172,31 @@ steps:
...
+    - task: AzureCLI@2
+      displayName: 'DotNet Test (AzureCLI)'
+      inputs:
+        azureSubscription: ${{ parameters.azureSubscription }}
+        scriptType: 'pscore'
+        scriptLocation: 'inlineScript'
+        inlineScript: 'dotnet test ${{ parameters.Solution }} --logger trx --results-directory $(Agent.TempDirectory) --configuration ${{ parameters.BuildConfiguration }} --no-restore --no-build'
+    
+    - task: PublishTestResults@2
```
Comment (WimVergouwe) [inline azure/pipelines/dotnet/ci.yml:185]: Shouldn't we add a succeededOrFailed() to both tasks? (see 'DotNet Test')

**Review by WimVergouwe — CHANGES_REQUESTED** [review]

File: azure/pipelines/dotnet/ci.yml (hunk 1 above)
Comment (chanw) [reply azure/pipelines/dotnet/ci.yml:23]: Something like this?

`  displayName: 'Optional Azure Subscription to run ''dotnet test'' under Azure CLI context. So you can access Azure resources (like App Configuration, Key Vault, etc) with authenticated credentials.''`

Or the explain in a comment (#)?

**Pull request merged**